pkg/agent/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -299,6 +299,18 @@ func (c *Controller) GetControllerConnectionStatus() bool {
 // and NetworkPolicies, and spawns workers that reconciles NetworkPolicy rules.
 // Run will not return until stopCh is closed.
 func (c *Controller) Run(stopCh <-chan struct{}) error {
+	klog.Info("Waiting for Antrea client to be ready")
+	if err := wait.PollImmediateUntil(200*time.Millisecond, func() (bool, error) {
+		if _, err := c.antreaClientProvider.GetAntreaClient(); err != nil {
+			return false, nil
+		}
+		return true, nil
+	}, stopCh); err != nil {
+		klog.Errorf("Unable to get Antrea client: %v", err)
+		return err
```

Comment:
Is this the right way to handle the error here? We will just log one error message and return the error (which will be ignored by the caller). I would argue that this is more of a "fatal" error, that should cause the agent to crash, so that it doesn't go unnoticed. What do you think?

Reply:
Thanks for pointing it out, I just took another look, `wait.PollImmediateUntil` only returns error when stopCh is closed, which is part of process exiting, then perhaps the log should just say it no longer waits for it and stop the goroutine gracefully. We shouldn't log errors in other controllers too.
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/shared_informer.go#L266
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/daemon/daemon_controller.go#L288